invokeai/app/api/routers/models.py

```diff
@@ -116,36 +80,50 @@ async def update_model(
     responses= {
         201: {"description" : "The model imported successfully"},
         404: {"description" : "The model could not be found"},
+        424: {"description" : "The model appeared to import successfully, but could not be found in the model manager"},
+        409: {"description" : "There is already a model corresponding to this path or repo_id"},
     },
     status_code=201,
     response_model=ImportModelResponse
 )
 async def import_model(
-        name: str = Query(description="A model path, repo_id or URL to import"),
-        prediction_type: Optional[Literal['v_prediction','epsilon','sample']] = Query(description='Prediction type for SDv2 checkpoint files', default="v_prediction"),
+        name: str = Body(description="A model path, repo_id or URL to import"),
+        prediction_type: Optional[Literal['v_prediction','epsilon','sample']] = \
+                Body(description='Prediction type for SDv2 checkpoint files', default="v_prediction"),
 ) -> ImportModelResponse:
     """ Add a model using its local path, repo_id, or remote URL """
+    
     items_to_import = {name}
     prediction_types = { x.value: x for x in SchedulerPredictionType }
     logger = ApiDependencies.invoker.services.logger
-    
-    installed_models = ApiDependencies.invoker.services.model_manager.heuristic_import(
-        items_to_import = items_to_import,
-        prediction_type_helper = lambda x: prediction_types.get(prediction_type)
-    )
-    if info := installed_models.get(name):
+
+    try:
+        installed_models = ApiDependencies.invoker.services.model_manager.heuristic_import(
+            items_to_import = items_to_import,
+            prediction_type_helper = lambda x: prediction_types.get(prediction_type)
+        )
+        info = installed_models.get(name)
+
+        if not info:
+            logger.error("Import failed")
+            raise HTTPException(status_code=424)
+        
         logger.info(f'Successfully imported {name}, got {info}')
         return ImportModelResponse(
```

Comment:
Same as `CreateModelResponse`, we already know the name (because we called the route with it) and it's also in the info object, and the response is never sent without the status of `success`. 

If `info` is a `model info` object, I think the response model should only include that.

Reply:
I've removed the status code (using HTTP status instead) and renamed "name", but have not otherwise changed the `ImportModelResponse`. The reason is that there is **not** a direct relationship between the model ID provided in the call and the model information retrieved. Consider what happens when you import a civitae URL. The input is:

```
{
  "location": "https://civitai.com/api/download/models/107366"
}
```
But the output is:

```
{
  "location": "https://civitai.com/api/download/models/107366",
  "info": {
    "name": "3DMM_V12.safetensors",
    "model_type": "lora",
    "base_model": "sd-1",
    "config": {
      "path": "models/sd-1/lora/3DMM_V12.safetensors",
      "description": "sd-1 lora model 3DMM_V12",
      "model_format": "lycoris",
      "error": null
    }
  }
}
```